src/kernels/radix.hpp

```diff
@@ -0,0 +1,105 @@
+/*******************************************************************************
+ *
+ * MIT License
+ *
+ * Copyright (c) 2024 Advanced Micro Devices, Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ *
+ *******************************************************************************/
+
+#ifndef GUARD_RADIX_H
+#define GUARD_RADIX_H
+
+#include <hip/hip_fp16.h>
+#include <hip/hip_runtime.h>
+
+#define RADIX_TYPE typename RadixType<DTYPE>::type
```

Comment:
I'd rather avoid defines and use explicit `using RType = typename RadixType<DTYPE>::type;` where it is required.

Reply:
Updated it.

Reply:
Now there is a discrepancy between `DTYPE` and `IN_OUT_TYPE`. 
`using` can be moved right inside `kthvalueFwd` function, like this: 
```C++
template <typename DTYPE>
__device__ void kthvalueFwd(DTYPE* input)
{
    using RADIX_TYPE = typename RadixType<DTYPE>::type;
    RADIX_TYPE val = 1;
    *input = val;
}
```

Reply:
Updated it.